radixdlt-java/src/main/java/com/radixdlt/client/core/network/epics/WebSocketsEpic.java

```diff
@@ -51,11 +55,24 @@ private WebSockets() {
 		 * @param node a radix node to get the websocket client for
 		 * @return a websocket client mapped to the node
 		 */
-		public WebSocketClient get(RadixNode node) {
-			return webSockets.computeIfAbsent(
-				node,
-				n -> new WebSocketClient(listener -> HttpClients.getSslAllTrustingClient().newWebSocket(n.getLocation(), listener))
-			);
+		public WebSocketClient getOrCreate(RadixNode node) {
+			synchronized (lock)	 {
```

Comment:
Might be simpler to use a ConcurrentHashMap and then use 
```
return webSockets.computeIfAbsent(node, node -> { 
  WebSocketClient newClient = new WebSocketClient(...);
  newNodes.onNext(newClient); 
  return newClient;
}
```

Reply:
Also, bonus points because we finally have a cache for this and don't keep spamming the nodes anymore. Haha.

Reply:
Yeah I originally had your solution but I didn't like the fact that `onNext` is called within the lock which can have bad perf consequences since it's not clear how long and where `onNext` goes.